client/signup/steps/domains/index.jsx

```diff
@@ -489,10 +489,9 @@ class DomainsStep extends Component {
 			showExampleSuggestions = true;
 		}
 
-		let includeWordPressDotCom = this.props.includeWordPressDotCom;
-		if ( 'undefined' === typeof includeWordPressDotCom ) {
-			includeWordPressDotCom = ! this.props.isDomainOnly;
-		}
+		const includeWordPressDotCom =
+			this.props.includeWordPressDotCom ??
+			( ! this.props.isDomainOnly && this.props.flowName !== 'onboarding-with-email' );
```

Comment:
I haven't done a thorough investigation, but it looks like we can define a new step to be utilized by the `onboarding-with-email` flow, rather than hardcoding the check here.

In `steps-pure.js`, the `domains` step is [configured](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/config/steps-pure.js#L342) as follows:
```
domains: {
			stepName: 'domains',
			apiRequestFunction: createSiteWithCart,
			providesDependencies: [
				'siteId',
				'siteSlug',
				'domainItem',
				'themeItem',
				'shouldHideFreePlan',
				'isManageSiteFlow',
			],
			optionalDependencies: [ 'shouldHideFreePlan', 'isManageSiteFlow' ],
			props: {
				isDomainOnly: false,
			},
			delayApiRequestUntilComplete: true,
		},
```

We can create a new step, say `mailbox-domain`,  based on `domains`, and configure `includeWordPressDotCom: false` as in `props`. (Note that the step name is reflected in the address bar, so at this step, the URL becomes `/start/onboarding-with-email/mailbox-domain`)

Then we can update the `onboarding-with-email` flow to replace the `domains` step with `mailbox-domain`:
https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/config/flows-pure.js#L158

Ideally, with a separate step defined, we can also customize the step title to say "Choose a domain for your Professional Email") by configuring the `stepSectionName` property. After a quick test, I see the the customize string does make its way to [getHeaderText()](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/domains/index.jsx#L633). However, there appears to e an issue in the `DomainsStep` where it doesn't render when a custom title is supplied as a `prop`. @delputnam Is there any reason why we are preventing the `DomainsStep` component from rendering when a custom title is supplied?

Reply:
Excellent feedback! 👍 Going with the approach you suggested.

I named the step `domains-with-email` in an attempt to align with how other `domain` variation steps are named. Sounds a bit awkward, but at least it's clear.

We can get back to the question about the header text in an upcoming PR.

Reply:
Actually, scratch that. It would make things easier to roll the update to the heading into this PR too.

Will update the PR once those changes are ready.